drivers/bme280/bme280.c

```diff
@@ -0,0 +1,415 @@
+    dev->calibration.dig_P9 = get_int16_le(buffer, BME280_DIG_P9_LSB_REG - offset);
+
+    dev->calibration.dig_H1 = buffer[BME280_DIG_H1_REG - offset];
+    dev->calibration.dig_H2 = get_int16_le(buffer, BME280_DIG_H2_LSB_REG - offset);
+    dev->calibration.dig_H3 = buffer[BME280_DIG_H3_REG - offset];
+    dev->calibration.dig_H4 = (((int16_t)buffer[BME280_DIG_H4_MSB_REG - offset]) << 4) +
+        (buffer[BME280_DIG_H4_H5_REG - offset] & 0x0F);
+    dev->calibration.dig_H5 = (((int16_t)buffer[BME280_DIG_H5_MSB_REG - offset]) << 4) +
+        ((buffer[BME280_DIG_H4_H5_REG - offset] & 0xF0) >> 4);
+    dev->calibration.dig_H6 = buffer[BME280_DIG_H6_REG - offset];
+
+    DEBUG("[INFO] Chip ID = 0x%02X\n", buffer[BME280_CHIP_ID_REG - offset]);
+
+    /* Config is only be writeable in sleep mode */
+    (void)i2c_write_reg(dev->params.i2c_dev, dev->params.i2c_addr,
+                        BME280_CTRL_MEAS_REG, 0);
+
+    uint8_t b;
+
+    /* Config Register */
+    /* spi3w_en unused */
+    b = ((dev->params.t_sb & 7) << 5) | ((dev->params.filter & 7) << 2);
+    write_u8_reg(dev, BME280_CONFIG_REG, b);
+
+    /*
+     * Note from the datasheet about ctrl_hum: "Changes to this register only become effective
+     * after a write operation to "ctrl_meas".
+     * So, set ctrl_hum first.
+     */
+    b = dev->params.humidOverSample & 7;
+    write_u8_reg(dev, BME280_CTRL_HUMIDITY_REG, b);
+
+    b = ((dev->params.tempOverSample & 7) << 5) |
+        ((dev->params.pressOverSample & 7) << 2) |
+        (dev->params.runMode & 3);
+    write_u8_reg(dev, BME280_CTRL_MEAS_REG, b);
+
+    return 0;
+}
+
+/**
+ * @brief Start a measument and read the registers
+ */
+static int do_measurement(bme280_t* dev)
+{
+    /*
+     * If settings has FORCED mode, then the device go to sleep after
+     * it finished the measurement. To read again we have to set the
+     * runMode back to FORCED.
+     */
+    uint8_t ctrl_meas = get_ctrl_meas(dev);
+    uint8_t runMode = ctrl_meas & 3;
+    if (runMode != dev->params.runMode) {
+        /* Set the runMode back to what we want. */
+        ctrl_meas &= ~3;
+        ctrl_meas |= dev->params.runMode;
+        write_u8_reg(dev, BME280_CTRL_MEAS_REG, ctrl_meas);
+
+        /* Wait for measurement ready? */
+        size_t count = 0;
+        while (count < 10 && (get_status(dev) & 0x08) != 0) {
+            ++count;
+        }
+        /* What to do when measuring is still on? */
+    }
+    int nr_bytes;
+    int nr_bytes_to_read = sizeof(measurement_regs);
+    uint8_t offset = BME280_PRESSURE_MSB_REG;
+
+    nr_bytes = i2c_read_regs(dev->params.i2c_dev, dev->params.i2c_addr,
+                             offset, measurement_regs, nr_bytes_to_read);
+    if (nr_bytes != nr_bytes_to_read) {
+        LOG_ERROR("Unable to read temperature data\n");
+        return -1;
+    }
+    dump_buffer("Raw Sensor Data", measurement_regs, nr_bytes);
+
+    return 0;
+}
+
+static uint8_t get_ctrl_meas(bme280_t* dev)
```

Comment:
I think this function is useless. It's only used once. I would prefer direct calls to the RIOT API : `i2c_read_reg`.
Same remark for `get_status`.